### PR TITLE
Reduce "perform updated on X ..." reconciler logs

### PR DESCRIPTION
**Pull request by kaovilai**
Reduce 🙈 update events and eliminate ObjectMeta update race 🏎 conditions for velero deployment and restic daemonset.

Now results from CreateOrUpdate are either created, or unchanged, update events are minimal.
No longer producing race condition update events thus reduces 99% of logs currently printed from these reconcilers.

### CODE REVIEW

**Comment by shubham-pampattiwar** [comment]
@kaovilai Do you mind adding a fix for this issue https://github.com/openshift/oadp-operator/issues/585 in this PR itself ?

**Comment by kaovilai** [comment]
> @kaovilai Do you mind adding a fix for this issue https://github.com/openshift/oadp-operator/issues/585 in this PR itself ?

Do you think it'll be too big. Could do in one pr

File: controllers/restic.go [hunk 1]
```diff
@@ -234,9 +230,59 @@ func (r *DPAReconciler) customizeResticDaemonset(dpa *oadpv1alpha1.DataProtectio
...
 	}
+
+	// Define defaults to avoid update events
+	// replace volumeSource HostPath Type if nil with empty string
+	setDsDefaults(ds, prevDs)
+
 	return ds, nil
 }
 
+func setDsDefaults(ds *appsv1.DaemonSet, prevDs *appsv1.DaemonSet) {
```
[the remark this answers is not in the document]
Comment (kaovilai) [reply controllers/restic.go:241]: `customizeVeleroDeployment` are already not producing excessive update events as is.

File: controllers/restic.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ const (
 
 var (
 	resticPvHostPath string = getResticPvHostPath()
+	emptyString             = ""
```
Comment (shubham-pampattiwar) [inline controllers/restic.go:34]: Please remove this as not in use after latest changes

File: controllers/restic.go [hunk 3]
```diff
@@ -144,21 +144,17 @@ func (r *DPAReconciler) buildResticDaemonset(dpa *oadpv1alpha1.DataProtectionApp
...
+		install.WithSecret(false)).Spec
 
-	ds.Name = resticDaemonSetName
-	ds.OwnerReferences = ownerRefs
-	return r.customizeResticDaemonset(dpa, ds)
+	return r.customizeResticDaemonset(dpa, ds, prevDs)
 }
 
-func (r *DPAReconciler) customizeResticDaemonset(dpa *oadpv1alpha1.DataProtectionApplication, ds *appsv1.DaemonSet) (*appsv1.DaemonSet, error) {
+func (r *DPAReconciler) customizeResticDaemonset(dpa *oadpv1alpha1.DataProtectionApplication, ds *appsv1.DaemonSet, prevDs *appsv1.DaemonSet) (*appsv1.DaemonSet, error) {
```
Comment (shubham-pampattiwar) [inline controllers/restic.go:157]: Revert to older function signature, as prevDS not needed now.

**Review by shubham-pampattiwar — CHANGES_REQUESTED** [review]

File: controllers/velero.go [hunk 4]
```diff
@@ -390,7 +390,7 @@ func (r *DPAReconciler) buildVeleroDeployment(veleroDeployment *appsv1.Deploymen
 		// see: https://github.com/vmware-tanzu/velero/blob/ed5809b7fc22f3661eeef10bdcb63f0d74472b76/pkg/install/deployment.go#L223-L261
 		// our secrets are appended to containers/volumeMounts in credentials.AppendPluginSpecificSpecs function
 		install.WithSecret(false),
-	)
+	).Spec
 	// adjust veleroDeployment from install
 	veleroDeployment.Name = deploymentName //reapply saved deploymentName and owner refs
```
Comment (shubham-pampattiwar) [inline controllers/velero.go:395]: @kaovilai Isn't this redundant ? as we are only taking the `spec` values from the install pkg and the metadata remains preserved.

Comment (kaovilai) [reply controllers/velero.go:395]: We were taking the whole thing.. objectmeta fields as well.. hence the race conditions.

Comment (shubham-pampattiwar) [reply controllers/velero.go:395]: I mean we do not need to reapply the name and owner refs here.

Comment (kaovilai) [reply controllers/velero.go]: yep.

**Comment by kaovilai** [comment]
Looks like there are stuff in objectMeta we would want to copy..
for veleroDeployment
- podAnnotations
.. so might as well do typemeta as well and delete TypeMeta test changes.

**Review by shubham-pampattiwar — APPROVED** [review]

**Pull request merged**